stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/MessageItem.kt

```diff
@@ -211,114 +304,194 @@ public fun DefaultMessageContainer(
+            .map { it.key }
+            .filter { supportedReactions[it] != null }
+            .map { type -> requireNotNull(supportedReactions[type]) to (type in ownReactions.map { it.type }) }
+
+        if (reactions.isNotEmpty()) {
+            MessageReactions(
+                modifier = modifier,
+                reactions = reactions
+            )
+        }
+    }
+}
 
-            if (ownsMessage) {
-                Spacer(modifier = Modifier.width(8.dp))
-            }
+/**
+ * Represents the default content shown at the bottom of the message list item.
+ *
+ * By default, the following can be shown in the footer:
+ * - uploading status
+ * - thread participants
+ * - message timestamp
+ *
+ * @param messageItem The message item to show the content for.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+public fun ColumnScope.DefaultMessageItemFooterContent(
+    messageItem: MessageItem,
+    modifier: Modifier = Modifier,
+) {
+    val message = messageItem.message
+    when {
+        message.isUploading() -> {
+            UploadingFooter(
+                modifier = modifier.align(End),
+                message = message
+            )
+        }
+        message.isDeleted() && messageItem.isMine -> {
+            OwnedMessageVisibilityContent(
+                modifier = modifier,
+                message = message
+            )
+        }
+        !message.isDeleted() -> {
+            MessageFooter(
+                messageItem = messageItem,
+                modifier = modifier
+            )
         }
     }
+
+    val position = messageItem.groupPosition
+    val spacerSize = if (position == None || position == Bottom) 4.dp else 2.dp
+
+    Spacer(Modifier.size(spacerSize))
 }
 
 /**
- * Represents the section of each message where the [Avatar] resides. In case we don't need to show an avatar, we instead
- * add a spacer, to keep messages aligned the same way.
+ * Represents the default content shown at the end of the message list item.
  *
- * @param position Position of the message in a group. This determines if we should show the user image or a spacer.
- * @param user The user that owns the message.
+ * By default, we show an extra spacing at the end of the message list item.
+ *
+ * @param messageItem The message item to show the content for.
+ * @param modifier Modifier for styling.
  */
 @Composable
-private fun RowScope.MessageAvatar(
-    position: MessageItemGroupPosition,
-    user: User,
+public fun DefaultMessageItemTrailingContent(
+    messageItem: MessageItem,
+    modifier: Modifier = Modifier,
 ) {
-    if (position == Bottom || position == None) {
-        UserAvatar(
-            modifier = Modifier
-                .padding(start = 8.dp, end = 8.dp)
-                .size(24.dp)
-                .align(Alignment.Bottom),
-            user = user,
-            showOnlineIndicator = false
-        )
+    if (messageItem.isMine) {
+        Spacer(modifier = modifier)
```

Comment:
This spacer will effectively be empty if the users don't add anything right? Can we maybe add a check where if the Modifier === Modifier, we don't add anything?